src/core/server/services/users/users.ts

```diff
@@ -781,7 +783,7 @@ export async function demoteUser(
     mongo,
     tenant.id,
     userID,
-    viewer.moderationScopes.siteIDs
+    siteIDs
```

Comment:
When we demote/promote with these siteIDs are we checking that the `viewer` (user doing the demote/promote) has all of the siteID's in their moderation scopes?

If we haven't done this elsewhere, maybe we want to check similar to the following?
```
if (
    isSiteModerationScoped(viewer.moderationScopes) &&
    !siteIDs.every((siteID) =>
      viewer.moderationScopes?.siteIDs?.includes(siteID)
    )
  ) {
    throw new Error("you are not permitted to promote the user on these sites");
  }
  ```

Reply:
This is checked in the `SiteModeratorActions` that controls which sites are shown as options for promotion/demotion. So, the `viewer` shouldn't be given the option to promote/demote on sites outside of their scope. But, I agree that it is also a good idea to also check this here just in case. Added...thanks!

Reply:
Cool, resolving and approving!